engine/render/src/render/material.cpp

```diff
@@ -571,4 +597,35 @@ namespace dmRender
         }
         return tag_count > 0; // don't render anything with no matches at all
     }
+
+    bool GetCanBindTexture(dmGraphics::HTexture texture, HSampler sampler, uint32_t unit)
+    {
+        dmGraphics::TextureType texture_type = dmGraphics::GetTextureType(texture);
+        Sampler* s = (Sampler*) sampler;
+
+        if (s == 0x0)
+        {
+            dmLogError("Unable to bind texture with type %s to a null sampler (texture unit %d).",
```

Comment:
Hm, this can get a bit spammy if you try to bind a texture to a slot that the shader doesn't have. Maybe we should log once? Or as a warning? It's not a dealbreaker since the material will still work 🤔